src/main/java/at/hannibal2/skyhanni/features/mining/glacitemineshaft/CorpseLocator.kt

```diff
@@ -0,0 +1,141 @@
+
+        EntityUtils.getAllEntities().filterIsInstance<EntityArmorStand>()
+            .filterNot { corpse ->
+                MineshaftWaypoints.waypoints.any { it.location.distance(corpse.getLorenzVec()) <= 3 }
+            }
+            .filter { entity ->
+                entity.showArms && entity.hasNoBasePlate() && !entity.isInvisible
+            }
+            .forEach { entity ->
+                val helmetDisplayName = StringUtils.stripControlCodes(entity.getCurrentArmor(3).displayName)
+                val corpseType = WaypointsType.entries.firstOrNull { it.helmetName == helmetDisplayName }
+                corpseType?.let { type ->
+                    for (offset in -1..3) {
+                        val canSee = entity.getLorenzVec().add(y = offset).canBeSeen()
+                        if (canSee) {
+                            MineshaftWaypoints.waypoints.add(Waypoint(waypointType = type, location = entity.getLorenzVec().add(y = 1), shared = false))
+                            ChatUtils.chat("Located a ${type.displayText} and marked its location with a waypoint.")
+                            break
+                        }
+                    }
+                }
+            }
+    }
+
+    private fun shareCorpse() {
+        MineshaftWaypoints.waypoints.filterNot { it.shared }
+            .filterNot { corpse ->
+                parsedLocations.any { corpse.location.distance(it) <= 5 }
+            }
+            .filter { it.location.distanceToPlayer() <= 5 }
+            .forEach { corpse ->
+                val location = corpse.location
+                val x = location.x.toInt()
+                val y = location.y.toInt()
+                val z = location.z.toInt()
+                val type = corpse.waypointType.displayText
+
+                HypixelCommands.partyChat("x: $x, y: $y, z: $z | ($type)")
+                corpse.shared = true
+            }
+    }
+
+    @Suppress("UNUSED_PARAMETER")
+    @SubscribeEvent
+    fun onWorldChange(event: LorenzWorldChangeEvent) {
+        if (parsedLocations.isNotEmpty()) parsedLocations.clear()
+    }
+
+    @Suppress("UNUSED_PARAMETER")
+    @SubscribeEvent
+    fun onSecondPassed(event: SecondPassedEvent) {
+        if (!isEnabled()) return
+
+        findCorpse()
+
+        if (!config.autoSendLocation) return
+        if (MineshaftWaypoints.waypoints.isEmpty()) return
+        if (PartyAPI.partyMembers.isEmpty()) return
+        shareCorpse()
+    }
+
+    @SubscribeEvent
+    fun onWorldRender(event: LorenzRenderWorldEvent) {
+        if (!IslandType.MINESHAFT.isInIsland()) return
+        if (MineshaftWaypoints.waypoints.isEmpty()) return
+
+        MineshaftWaypoints.waypoints.forEach {
+            event.drawWaypointFilled(it.location, it.waypointType.color.toColor(), seeThroughBlocks = true)
+            event.drawDynamicText(it.location, "§e${it.waypointType.displayText}", 1.0)
+        }
+    }
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
```

Comment:
dont use chat event, use player chat events (party + all chat) instead. this will reduce the regex complexity drastically